homeassistant/components/zwave_js/discovery_data_template.py

```diff
@@ -107,3 +175,52 @@ def current_temperature_value(resolved_data: dict[str, Any]) -> ZwaveValue | Non
             return lookup_table.get(lookup_key)
 
         return None
+
+
+class NumericSensorDataTemplate(BaseDiscoverySchemaDataTemplate):
+    """Data template class for Z-Wave Sensor entities."""
+
+    def resolve_data(self, value: ZwaveValue) -> dict[str, Any]:
+        """Resolve helper class data for a discovered value."""
+        data = {}
+
+        if value.command_class == CommandClass.BATTERY:
+            data[ATTR_ENTITY_DESC_KEY] = ENTITY_DESC_KEY_BATTERY
+        elif value.command_class == CommandClass.METER:
+            cc_specific = value.metadata.cc_specific
```

Comment:
I think I am going to move the logic to get the `MeterType` and `ScaleType` into the library because then it can be reused. I can also then remove the `MeterType` from the discovery template data so it's consistent across `CommandClass` types